atlasdb-ete-tests/src/test/java/com/palantir/atlasdb/ete/CassandraMultinodeEteTest.java

```diff
@@ -0,0 +1,228 @@
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+
+import com.google.common.base.Optional;
+import com.google.common.collect.ImmutableList;
+import com.jayway.awaitility.Awaitility;
+import com.palantir.atlasdb.http.AtlasDbHttpClients;
+import com.palantir.atlasdb.todo.ImmutableTodo;
+import com.palantir.atlasdb.todo.Todo;
+import com.palantir.atlasdb.todo.TodoResource;
+import com.palantir.docker.compose.DockerComposeRule;
+import com.palantir.docker.compose.connection.Container;
+import com.palantir.docker.compose.connection.DockerPort;
+import com.palantir.docker.compose.connection.waiting.HealthCheck;
+import com.palantir.docker.compose.connection.waiting.SuccessOrFailure;
+import com.palantir.docker.compose.execution.DockerComposeExecArgument;
+import com.palantir.docker.compose.execution.DockerComposeExecOption;
+import com.palantir.giraffe.command.Command;
+import com.palantir.giraffe.command.CommandResult;
+import com.palantir.giraffe.command.Commands;
+import com.palantir.giraffe.host.HostAccessors;
+import com.palantir.giraffe.host.HostControlSystem;
+
+public class CassandraMultinodeEteTest {
+
+    private static final Gradle GRADLE_PREPARE_TASK =
+            Gradle.ensureTaskHasRun(":atlasdb-ete-test-utils:buildCassandraImage");
+
+    private static final int TIMELOCK_SERVER_PORT = 3828;
+
+    private static final String CONTAINER_LOGS_DIRECTORY = "container-logs/cassandra-multinode";
+
+    private static final List<String> CASSANDRA_NODES = ImmutableList.of("cassandra1", "cassandra2", "cassandra3");
+
+    private static final long MAX_CASSANDRA_NODE_DOWN_MILLIS = 30000;
+    private static final long MAX_CASSANDRA_NODES_RUNNING_MILLIS = 3000;
+
+    private static final DockerComposeRule MULTINODE_CASSANDRA_SETUP = DockerComposeRule.builder()
+            .file("docker-compose.multiple-cassandra.yml")
+            .waitingForService("cassandra1", Container::areAllPortsOpen)
+            .waitingForService("cassandra2", Container::areAllPortsOpen)
+            .waitingForService("cassandra3", Container::areAllPortsOpen)
+            .waitingForService("ete1", toBeReady())
+            .saveLogsTo(CONTAINER_LOGS_DIRECTORY)
+            .build();
+
+
+    @ClassRule
+    public static final RuleChain PREPARED_DOCKER_SETUP = RuleChain
+            .outerRule(GRADLE_PREPARE_TASK)
+            .around(MULTINODE_CASSANDRA_SETUP);
+
+    @Test
+    public void shouldRunTransactionsWithAllCassandraNodesRunningWithoutUnacceptableDelay()
+            throws InterruptedException, IOException {
+        TodoResource clientToSingleNode = createClientFor(TodoResource.class, asPort("ete1"));
+
+        long transactionStartTime = System.currentTimeMillis();
+        assertAddTodoTransactionWasSuccessful(clientToSingleNode);
+        long transactionEndTime = System.currentTimeMillis();
+
+        long transactionTimeWithAllNodesRunning = transactionEndTime - transactionStartTime;
```

Comment:
Can we extract this duplicated timing code? Maybe a utility function like 

```
    private long timeit(Runnable task) {
        long transactionStartTime = System.currentTimeMillis();
        task.run();
        long transactionEndTime = System.currentTimeMillis();
        return transactionEndTime - transactionStartTime;
    }
```

and then use
```
long transactionTimeAfterNodeIsKilled = timeit(() -> assertAddTodoTransactionWasSuccessful(clientToSingleNode));
```